app/models/translation.rb

```diff
@@ -94,18 +95,40 @@ def prevent_destroy_published
   def push_published_to_s3
     return unless is_published
 
-    name_desc_onesky if resource.uses_onesky?
+    ActiveRecord::Base.transaction do
+      if resource.uses_onesky?
+        phrases = manifest_translated_phrases
+        name_desc_onesky(phrases)
+        create_translated_attributes(phrases)
```

Comment:
hmm, these could still leak partial attributes if `Package.new(self).push_to_s3` fails. But I'm not sure of a great solution to this because the `push_to_s3` process shouldn't be part of the transaction.

For now this is fine unless you have a potential quick solution in mind